scrapers/ms/events.py

```diff
@@ -4,27 +4,119 @@
 from utils.events import match_coordinates
 from openstates.scrape import Scraper
 from openstates.scrape import Event
+from spatula import PdfPage, HtmlPage, URL, XPath
+import datetime
 
+TZ = pytz.timezone("US/Central")
 
-class MSEventScraper(Scraper):
-    _tz = pytz.timezone("US/Central")
-    chamber_abbrs = {"upper": "s", "lower": "h"}
-    chamber_names = {"upper": "Senate", "lower": "House"}
+start_time_re = re.compile(
+    r"^(MONDAY|TUESDAY|WEDNESDAY|THURSDAY|FRIDAY|SATURDAY|SUNDAY)", flags=re.IGNORECASE
+)
+page_number_re = re.compile(r"^page \d+$", flags=re.IGNORECASE)
+
+# Bills have a ". " after each letter in their acronym, so regex is a little long
+bill_re = re.compile(
+    r"(S\.? ?C\.? ?|S\.? ?N\.? ?|H\.? ?B\.? ?|H\.? ?R\.? ?|S\.? ?B\.? ?|J\.? ?R\.? ?|H\.? ?C\.? ?|S\.? ?R\.? ?).{0,6}?(\d+)"
+)
+
+
+# Finds the required agenda pdf, the url changes yearly
+class SenateAgenda(HtmlPage):
+    source = (
+        "http://www.legislature.ms.gov/calendars-and-schedules/senate-committee-agenda/"
+    )
+
+    def process_page(self):
+        pdf_link = XPath("//h3/a/@href").match_one(self.root)
+        yield from SenateAgendaPdf(source=URL(pdf_link)).do_scrape()
+
+
+# Parses events from a pdf
+class SenateAgendaPdf(PdfPage):
+    def process_page(self):
+        event = None
 
+        # Strip all lines and remove empty lines
+        lines = [line.strip() for line in self.text.splitlines() if line.strip() != ""]
```

Comment:
Very minor stylistic change, but this can be modified to the more Pythonic:
```
lines = [line.strip() for line in self.text.splitlines() if line.strip()]
```
- given that empty strings are false-y.

Reply:
That's good to know - thank you. Changed with https://github.com/openstates/openstates-scrapers/pull/4541/commits/1e7ce3f30257842675348a025625519129e93c19